metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_api.py

```diff
@@ -673,55 +674,116 @@ def get_description_for_resource(self, resource_id: str) -> Optional[str]:
             )
         return None
 
-    def get_containers_for_location(
-        self, resource_id: str, path: List[str]
-    ) -> List[Dict[str, str]]:
-        containers = []
+    def _check_pattern_match(
+        self,
+        pattern: str,
+        paths: List[str],
+        full_path: str,
+    ) -> bool:
+        """Helper method to check if a pattern matches any of the paths."""
+        # Handle exact pattern matches
+        if re.search(f"^{pattern}$", full_path, re.IGNORECASE):
```

Comment:
Have you considered using simply `self.config.schema_pattern.is_allowed()` ?

We use this everywhere even for hierarchical representations. This usually suffices for all cases once its clear what the regex matches against and by using a few wildcards `.*` if filtering using subpaths.

This would make this a lot simpler, predictable , at the same time, consistent with rest of code base,

Reply:
As per our discussion with Dremio's APIs only returning parts of the path at one time (e.g. returning `a` then `b` as `a` child of `b` then `c` as a child of `b` in an `a.b.c` schema) this needs to be handled separately. This has now been updated to match the implementation of `schema_pattern.is_allowed()` but the implementation will need to be custom for Dremio's connector due to the way that the API returns the path parts.